src/components/filters/MeasureValueFilter/Dropdown.tsx

```diff
@@ -0,0 +1,121 @@
+import { DropdownBody } from "./DropdownBody";
+import { getOperatorTranslationKey } from "./utils";
+
+export interface IDropdownButtonProps {
+    isActive?: boolean;
+    onClick: (e: React.SyntheticEvent) => void;
+    measureTitle?: string;
+    operator?: string;
+    operatorTitle?: string;
+    value?: IValue;
+}
+
+export interface IDropdownOwnProps {
+    button?: React.ComponentType<IDropdownButtonProps>;
+    onApply: (operator: MeasureValueFilterConditionOperator, value?: IValue) => void;
+    locale?: string;
+    measureTitle?: string;
+    operator?: MeasureValueFilterConditionOperator;
+    value?: IValue;
+    displayDropdown?: false;
+}
+
+export type IDropdownProps = IDropdownOwnProps & InjectedIntlProps;
+
+interface IDropdownState {
+    displayDropdown: boolean;
+}
+
+class DropdownWrapped extends React.PureComponent<IDropdownProps, IDropdownState> {
+    public static defaultProps: Partial<IDropdownOwnProps> = {
+        button: DefaultDropdownButton,
+        value: {},
+    };
+
+    private toggleButtonRef: EventTarget = null;
+
+    constructor(props: IDropdownProps) {
+        super(props);
+
+        const { displayDropdown } = props;
+
+        this.state = {
+            displayDropdown,
+        };
+    }
+
+    public render() {
+        const { intl, button: DropdownButton, measureTitle, operator, value, locale } = this.props;
+        const { displayDropdown } = this.state;
+
+        const operatorTitle = intl.formatMessage({
+            id: getOperatorTranslationKey(operator ? operator : "ALL"),
+        });
+
+        return (
+            <>
+                <DropdownButton
+                    isActive={displayDropdown}
+                    onClick={this.toggle}
+                    measureTitle={measureTitle}
+                    operatorTitle={operatorTitle}
+                    operator={operator ? operator : "ALL"}
+                    value={value}
+                />
+                {displayDropdown ? (
+                    <Overlay
+                        alignTo={this.toggleButtonRef}
+                        alignPoints={[{ align: "bl tl" }]}
+                        closeOnOutsideClick={true}
+                        closeOnParentScroll={true}
+                        closeOnMouseDrag={true}
+                        onClose={this.closeDropdown}
+                    >
+                        <DropdownBody
+                            operator={operator}
+                            value={value}
+                            locale={locale}
+                            onCancel={this.closeDropdown}
+                            onApply={this.onApply}
+                        />
+                    </Overlay>
+                ) : null}
+            </>
+        );
+    }
+
+    private toggle = (e: React.SyntheticEvent) => {
+        this.toggleButtonRef = !this.state.displayDropdown ? e.currentTarget : null;
+
+        this.setState({ displayDropdown: !this.state.displayDropdown });
```

Comment:
This might cause concurrency issues, please use the callback syntax (see [docs](https://reactjs.org/docs/react-component.html#setstate) towards the end of the setState section):
```jsx
this.setState(state => ({ ...state, displayDropdown: !state.displayDropdown });
```